src/components/DatePicker/DatePicker.tsx

```diff
@@ -0,0 +1,199 @@
+  onChange,
+}) => {
+  const theme = useTheme();
+  const dayPickerRef = useRef<DayPickerInput>(null);
+  const daysInitialState = { from: undefined, to: undefined };
+  const [selectedOption, setSelectedOption] = useState();
+  const [selectedDay, setSelectedDay] = useState<DateRange>({ from: undefined, to: undefined });
+
+  useEffect(() => {
+    if (onChange) {
+      onChange(selectedDay);
+    }
+  }, [onChange, selectedDay]);
+
+  const handleDayRangeClick = useCallback(
+    day => {
+      const aboutToCompleteBothDates = selectedDay.from && !selectedDay.to;
+      const range = DateUtils.addDayToRange(
+        day,
+        // @TODO problem with the library as they define RangeModifiers as not null or undefined when they use it like this
+        // @ts-ignore
+        aboutToCompleteBothDates ? selectedDay : daysInitialState
+      );
+
+      if (aboutToCompleteBothDates) {
+        dayPickerRef.current?.hideDayPicker();
+      }
+
+      setSelectedDay(range);
+      setSelectedOption('custom');
+    },
+    [selectedDay, setSelectedDay, setSelectedOption, dayPickerRef.current]
+  );
+
+  const handleDayClick = useCallback(
+    day => {
+      setSelectedDay({ from: day, to: day });
+      setSelectedOption('custom');
+      dayPickerRef.current?.hideDayPicker();
+    },
+    [setSelectedDay, setSelectedOption, dayPickerRef.current]
+  );
+
+  const handleSelectedOptions = (option: string) => {
+    const foundOption = extraOptions.find(optionItem => optionItem.value === option);
+    dayPickerRef.current?.hideDayPicker();
+
+    if (foundOption) {
+      setSelectedDay(
+        Array.isArray(foundOption.dates)
+          ? { from: foundOption.dates[0], to: foundOption.dates[1] }
+          : { from: foundOption.dates, to: foundOption.dates }
+      );
+    }
+
+    setSelectedOption(option);
+  };
+
+  const modifiers = { start: selectedDay.from, end: selectedDay.to };
+  const getDateFormatted = useCallback(
+    (date: Date | undefined) => (date ? dayjs(date).format('MM/DD/YYYY') : ''),
+    []
+  );
+
+  return (
+    <div css={datePickerStyles({ isRangePicker })(theme)}>
+      <DayPickerInput
+        ref={dayPickerRef}
+        overlayComponent={(props: DayPickerInputProps) => (
+          <OverlayComponent
+            selectedOption={selectedOption}
+            setSelectedOption={handleSelectedOptions}
+            extraOptions={extraOptions}
+            isRangePicker={isRangePicker}
+            {...props}
+          />
+        )}
+        dayPickerProps={{
+          onDayClick: isRangePicker ? handleDayRangeClick : handleDayClick,
+          selectedDays: selectedDay as RangeModifier,
+          modifiers: isRangePicker ? modifiers : undefined,
+          firstDayOfWeek: 1,
+          numberOfMonths: isRangePicker ? 2 : 1,
+          disabledDays: disableFutureDates
+            ? [
+                {
+                  after: new Date(),
+                },
+              ]
+            : undefined,
+        }}
```

Comment:
You can assign this to a variable to reduce length of the rendered JSX code. I think, it will be easier for someone to read the final rendered JSX.

example: 
```
return <SomeComponent  dayPickerProps={{
          onDayClick: isRangePicker ? handleDayRangeClick : handleDayClick,
          selectedDays: selectedDay as RangeModifier,
          modifiers: isRangePicker ? modifiers : undefined,
          firstDayOfWeek: 1,
          numberOfMonths: isRangePicker ? 2 : 1,
          disabledDays: disableFutureDates
            ? [
                {
                  after: new Date(),
                },
              ]
            : undefined,
        }}/>
```
vs
```
return <SomeComponent dayPickerProps={dayPickerProps}/>
```